src/platform_impl/linux/wayland/seat/touch/handlers.rs

```diff
@@ -0,0 +1,124 @@
+            let touch_point = match inner.touch_points.iter().find(|p| p.id == id) {
+                Some(touch_point) => touch_point,
+                None => return,
+            };
+
+            let scale_factor = sctk::get_surface_scale_factor(&touch_point.surface) as f64;
+            let location = touch_point.position.to_physical(scale_factor);
+            let window_id = wayland::make_wid(&touch_point.surface);
+
+            event_sink.push_window_event(
+                WindowEvent::Touch(crate::event::Touch {
+                    device_id: crate::event::DeviceId(crate::platform_impl::DeviceId::Wayland(
+                        DeviceId,
+                    )),
+                    phase: TouchPhase::Ended,
+                    location,
+                    force: None, // TODO
+                    id: id as u64,
+                }),
+                window_id,
+            );
+        }
+        TouchEvent::Motion { id, x, y, .. } => {
+            let touch_point = match inner.touch_points.iter_mut().find(|p| p.id == id) {
+                Some(touch_point) => touch_point,
+                None => return,
+            };
+
+            touch_point.position = LogicalPosition::new(x, y);
+
+            let scale_factor = sctk::get_surface_scale_factor(&touch_point.surface) as f64;
+            let location = touch_point.position.to_physical(scale_factor);
+            let window_id = wayland::make_wid(&touch_point.surface);
+
+            event_sink.push_window_event(
+                WindowEvent::Touch(crate::event::Touch {
+                    device_id: crate::event::DeviceId(crate::platform_impl::DeviceId::Wayland(
+                        DeviceId,
+                    )),
+                    phase: TouchPhase::Moved,
+                    location,
+                    force: None, // TODO
+                    id: id as u64,
+                }),
+                window_id,
+            );
+        }
+        TouchEvent::Frame => (),
+        TouchEvent::Cancel => {
+            for touch_point in inner.touch_points.drain(..) {
+                let scale_factor = sctk::get_surface_scale_factor(&touch_point.surface) as f64;
+                let location = touch_point.position.to_physical(scale_factor);
+                let window_id = wayland::make_wid(&touch_point.surface);
+
+                event_sink.push_window_event(
+                    WindowEvent::Touch(crate::event::Touch {
+                        device_id: crate::event::DeviceId(crate::platform_impl::DeviceId::Wayland(
+                            DeviceId,
+                        )),
+                        phase: TouchPhase::Cancelled,
+                        location,
+                        force: None, // TODO
+                        id: touch_point.id as u64,
+                    }),
+                    window_id,
+                );
+            }
+        }
+        _ => {
+            // This space intentionally left blank.
+        }
```

Comment:
Ofc.

Reply:
Makes sense. Is this something that could be perhaps included into sctk? If you add this just to get non-exhaustive matching that is.

Reply:
>Makes sense. Is this something that could be perhaps included into sctk? If you add this just to get non-exhaustive matching that is.

For things that are directly from SCTK -> yes, from things that are from wayland-client -> No. The keyboard helpers are provided by SCTK, and so you don't have non exhausitve enum for events there, however touch and pointer events, as well as text input, are from wayland-client, and this code is generated by wayland-scanner, so you can't hide them. If you add helpers into sctk for those events, maybe, but there's not point in helpers in sctk for those things, since they are simple.

Reply:
Why can't the non-exhaustive argument be added to wayland-client?

Reply:
The non-exhaustive is there and presented on every enum `non-exhaustive`, so I'm not sure what you're talking about? Do you ask why keyboard don't have `non-exhaustive`? Because there's no need to.

Reply:
Or do you mean that it should have some special `Other(something)`? If so we may ask in e.g. `#smithay` way it's that way, I'd assume that it's not trivial to generate such thing, since you have arbitrary amount of args.

Reply:
Instead of using `_ => (),` here, you can just mark the thing your matching upstream as non exhaustive using the Rust attribute?

Reply:
Could you link to docs wrt what you mean, I mean the enum upstream is `non-exhaustive` and so Rust asks me to put `_ => ()` here.

Reply:
>  I mean the enum upstream is non-exhaustive and so Rust asks me to put _ => () here.

Ah, that's all I wanted to know. Then there's no reason to comment that it is, it should be obvious due to compiler telling you it is?

Reply:
>Then there's no reason to comment that it is, it should be obvious due to compiler telling you it is?

I just don't want to make this as a one liner due to rustfmt without bracers. I can remove that if you don't like that style.

Reply:
I'm not sure what you mean? What does rustfmt do?

Reply:
It'll format to  `_ => (),` from  `_ => { /* new line */ }`, since unless you have some comment inside the bracers and I wanted the later style, I can use former style, I don't care that much.

Reply:
You mean it'll format into this?

```rust
_ => {
}
```

Reply:
Yeah, I just don't like how it looks, though it's just me. I'll update those.

Reply:
Everything else is much worse, so `_ => (),` is really the only acceptable option here.